lib/core/uri_parser.js

```diff
@@ -687,6 +687,8 @@ function parseConnectionString(uri, options, callback) {
     return callback(new MongoParseError('directConnection option requires exactly one host'));
   }
 
+  parsedOptions._directConnection = Boolean(parsedOptions.directConnection);
```

Comment:
Open changing the naming here, we need a way to know if `directConnect` is defaulted to `true` or not.